shell/common/node_bindings.cc

```diff
@@ -476,25 +475,33 @@ node::Environment* NodeBindings::CreateEnvironment(
     // not to register its handler (overriding blinks) in non-browser processes.
     flags |= node::EnvironmentFlags::kNoRegisterESMLoader |
              node::EnvironmentFlags::kNoInitializeInspector;
-    v8::TryCatch try_catch(context->GetIsolate());
-    env = node::CreateEnvironment(
-        isolate_data_, context, args, exec_args,
-        static_cast<node::EnvironmentFlags::Flags>(flags));
-    DCHECK(env);
-
-    // This will only be caught when something has gone terrible wrong as all
-    // electron scripts are wrapped in a try {} catch {} by webpack
-    if (try_catch.HasCaught()) {
-      LOG(ERROR) << "Failed to initialize node environment in process: "
-                 << process_type;
+  }
+
+  v8::TryCatch try_catch(isolate);
+  env = node::CreateEnvironment(
+      isolate_data_, context, args, exec_args,
+      static_cast<node::EnvironmentFlags::Flags>(flags));
+
+  if (try_catch.HasCaught()) {
+    std::string err_msg =
+        "Failed to initialize node environment in process: " + process_type;
+    v8::Local<v8::Value> exception = try_catch.Exception();
+    if (!exception->IsNull() && exception->IsObject()) {
+      v8::Local<v8::Value> error_message = try_catch.Exception();
+      v8::MaybeLocal<v8::Value> maybe_message = exception.As<v8::Object>()->Get(
+          context, gin::ConvertToV8(isolate, "message"));
+      std::string msg;
+      if (maybe_message.ToLocal(&error_message) && error_message->IsString()) {
```

Comment:
```suggestion
      v8::MaybeLocal<v8::Value> maybe_message = exception.As<v8::Object>()->Get(
          context, gin::ConvertToV8(isolate, "message"));
      v8::Local<v8::Value> error_message;
      if (maybe_message.ToLocal(&error_message) && error_message->IsString()) {
        std::string msg;
```
* there's no need to assign `try_catch.Exception()` to the `error_message` local first if we're just going to change its value to the ToLocal of `maybe_message` right afterwards
* moved the `msg` string closer to where it is actually required